php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -171,10 +170,13 @@ class NDB_BVL_Battery
     {
         $DB = Database::singleton();
         // assert that a battery has already been selected
-        $error = $this->_assertBatterySelected();
+        $this->_assertBatterySelected();
 
         // assert that the instrument is valid
         $error = $this->_assertValidInstrument($testName);
+        if ($error !== true) {
```

Comment:
for some reason assertBatterySelected and assertValidInstrument behave differently. assertBatterySelected throws an error, assertValidInstrument returns false.

The return value wasn't checked for assertValidInstrument, so without turning this to an exception the assertion doesn't do anything.